modules/data.atmosphere/R/download.ECMWF.R

```diff
@@ -0,0 +1,201 @@
+      conditionMessage(e)
+    )
+  })
+  
+  hour <- fchour
+  
+  all_hours <- c(00, 12)
+  
+  if (any(!hour %in% all_hours)) {
+    bad_hours <- setdiff(hour, all_hours)
+    PEcAn.logger::logger.severe(sprintf(
+      "Invalid forecast hours  %s. 15 day forecast hours must be one of the following: %s",
+      paste0("`", bad_hours, "`", collapse = ", "),
+      paste0("`", all_hours, "`", collapse = ", ")
+    ))
+  }
+  
+  if (fchour == 00){
+    time <- 0
+  } else if (fchour == 12){
+    time <- 12
+  }
+  
+  all_parameters <- c("2t", "tp", "10u", "10v", "q", "r", "sp")
+  
+  if (tolower(parameters) == "all") {
+    parameter_types <- all_parameters
+  }
+  
+  if (any(!parameter_types %in% all_parameters)) {
+    bad_parameters <- setdiff(parameter_types, all_parameters)
+    PEcAn.logger::logger.severe(sprintf(
+      "Invalid parameter types %s. Products must be one of the following: %s",
+      paste0("`", bad_parameters, "`", collapse = ", "),
+      paste0("`", all_parameters, "`", collapse = ", ")
+    ))
+  }
+  
+  if (stream == "enfo"){
+    time <- time
+    step <- 360
+    type <- type
+  }  else{ 
+    PEcAn.logger::logger.severe(sprintf(
+      "Invalid data stream. Currently, data stream `enfo` is supported."
+    ))
+  }
+  
+  # Python script "download_ecmwf.py" to get latest forecast date and download forecast datasets
+  script.path = file.path(system.file("ECMWF/download_ecmwf_opendata.py", package = "PEcAn.data.atmosphere"))
+  reticulate::source_python(script.path)
+  
+  date_latestdata <- ecmwflatest(time, step, stream, type, all_parameters)
+  latest_filedate <- strtoi(gsub("-", "", substr(as.character.Date(date_latestdata), 1, 10)))
+  
+  current_date <- strtoi(gsub("-", "", Sys.Date()))
+  
+  date <- if (latest_filedate == current_date){
+    # today
+    0
+  } else if ((latest_filedate - current_date) == -1){
+    # yesterday
+    -1
+  } else if ((latest_filedate - current_date) == -2){
+    # the day before yesterday
+    -2
+  } else { PEcAn.logger::logger.severe(sprintf(
+    "Invalid file download date."
+  ))
+  }
+  
+  # Removes any already existing files with base name same as `latest_filedate``
+  if ((length(list.files(path = outfolder, pattern = as.character(latest_filedate)) ) > 0) == TRUE){
+    file.remove(list.files(outfolder, pattern = as.character(latest_filedate)))
+  }
+  
+  fname <- paste(latest_filedate, time, step, stream, type, sep = "_")
+  
+  in_filename <- paste0(fname, ".grib2")
+  
+  data_download <- ecmwfdownload(date, time, step, stream, type, all_parameters, in_filename)
+  
+  # Python script "ecmwf_grib2nc.py" to convert grib2 to netCDF
+  # Converting Grib2 file to ensemble wise NetCDF file
+  script.path = file.path(system.file("ECMWF/ecmwf_grib2nc.py", package = "PEcAn.data.atmosphere"))
+  reticulate::source_python(script.path)
+  
+  nc_ecmwf <- grib2nc_ecmwf(in_filename, outfolder, out_filename)
```

Comment:
`out_filename` is not specified anywhere in this function

Reply:
in addition, looking at `grib2nc_ecmwf` this function call is missing 2 required args '`lat_in`' and '`lon_in`'